cachi2/core/resolver.py

```diff
@@ -0,0 +1,30 @@
+from typing import Callable
+
+from cachi2.core.errors import UnsupportedFeature
+from cachi2.core.models.input import PackageManagerType, Request
+from cachi2.core.models.output import RequestOutput, merge_outputs
+from cachi2.core.package_managers import gomod
+
+Handler = Callable[[Request], RequestOutput]
+
+_package_managers: dict[PackageManagerType, Handler] = {
+    "gomod": gomod.fetch_gomod_source,
+}
+
+
+supported_package_managers = list(_package_managers)
+
+
+def resolve_packages(request: Request) -> RequestOutput:
+    """Run all requested package managers, return their combined output."""
+    # de-dupe but preserve order
+    requested_types = dict.fromkeys(pkg.type for pkg in request.packages).keys()
+    unsupported_types = requested_types - _package_managers.keys()
+    if unsupported_types:
+        raise UnsupportedFeature(
+            f"Package manager(s) not yet supported: {', '.join(sorted(unsupported_types))}",
+            # unknown package managers shouldn't get past input validation
+            solution="But the good news is that we're already working on it!",
```

Comment:
"But the good news is that YOU can work on it." :laughing: 

On a serious note, an ideia is to point them to the part of the README that shows the plans for package manager support (aka what CachiTO supports).

Reply:
The thing is, the only way you get this error message is if we already have that package manager defined as a valid input (i.e. already working on it)

Reply:
Should I try to allow arbitrary strings as the package manager to make sure it only fails here?

Reply:
No such luck for the Union-based solution. Best I could come up with was this

```diff
diff --git a/cachi2/core/models/input.py b/cachi2/core/models/input.py
index 47d48c1..3300b17 100644
--- a/cachi2/core/models/input.py
+++ b/cachi2/core/models/input.py
@@ -53,7 +53,7 @@ Flag = Literal["cgo-disable", "force-gomod-tidy", "gomod-vendor", "gomod-vendor-
 class _PackageInputBase(pydantic.BaseModel, extra="forbid"):
     """Common input attributes accepted for all package types."""
 
-    type: PackageManagerType
+    type: str
     path: Path = Path(".")
 
     @pydantic.validator("path")
@@ -87,8 +87,14 @@ class PipPackageInput(_PackageInputBase):
         return check_sane_relpath(path)
 
 
+class UnsupportedPackageInput(_PackageInputBase):
+    """TODO."""
+
+    type: Literal["npm", "yarn", "rubygems"]
+
+
 PackageInput = Annotated[
-    Union[GomodPackageInput, PipPackageInput],
+    Union[GomodPackageInput, PipPackageInput, UnsupportedPackageInput],
     # https://pydantic-docs.helpmanual.io/usage/types/#discriminated-unions-aka-tagged-unions
     pydantic.Field(discriminator="type"),
 ]
```